go/logic/migrator.go

```diff
@@ -219,7 +219,11 @@ func (this *Migrator) onChangelogStateEvent(dmlEvent *binlog.BinlogDMLEvent) (er
 // listenOnPanicAbort aborts on abort request
 func (this *Migrator) listenOnPanicAbort() {
 	err := <-this.migrationContext.PanicAbort
+
+	//should cleanup before exit.
+	this.finalCleanup()
```

Comment:
How do you do in github, when fail, the dba check the reason and clean up? critical error may b often.